adapters/pubmatic/pubmatic.go

```diff
@@ -25,36 +26,61 @@ type PubmaticAdapter struct {
 	URI  string
 }
 
-// used for cookies and such
+/* Name - export adapter name */
 func (a *PubmaticAdapter) Name() string {
 	return "pubmatic"
 }
 
+// used for cookies and such
+func (a *PubmaticAdapter) FamilyName() string {
+	return "pubmatic"
+}
+
 func (a *PubmaticAdapter) SkipNoCookies() bool {
 	return false
 }
 
+// Below is bidder specific parameters for pubmatic adaptor,
+// PublisherId and adSlot are mandatory parameters, others are optional parameters
+// Keywords is bid specific parameter,
+// WrapExt needs to be sent once per bid request
 type pubmaticParams struct {
-	PublisherId string `json:"publisherId"`
-	AdSlot      string `json:"adSlot"`
+	PublisherId string            `json:"publisherId"`
+	AdSlot      string            `json:"adSlot"`
+	WrapExt     json.RawMessage   `json:"wrapper,omitempty"`
+	Keywords    map[string]string `json:"keywords,omitempty"`
 }
 
+const (
+	INVALID_PARAMS    = "Invalid BidParam"
+	MISSING_PUBID     = "Missing PubID"
+	MISSING_ADSLOT    = "Missing AdSlot"
+	INVALID_WRAPEXT   = "Invalid WrapperExt"
+	INVALID_ADSIZE    = "Invalid AdSize"
+	INVALID_WIDTH     = "Invalid Width"
+	INVALID_HEIGHT    = "Invalid Height"
+	INVALID_MEDIATYPE = "Invalid MediaType"
+	INVALID_ADSLOT    = "Invalid AdSlot"
+)
```

Comment:
All these constants you use exactly one time. Do you anticipate that your error message may change, or you have some other reason to store them as constants rather than hardcoding them into the message strings? Not that it matters, just more curious.

Reply:
Currently we are using the constants only one time, but we have kept them as constants for future use so that we use same error messages through out the adapter code.